### PR TITLE
BTE-128: add arch docs of re-design

### PR DESCRIPTION
**Pull request by zhihuizhangjaden**
## Context

* Describe the changes made and why they were made.
* Provide links to any issues or designs for this change.

## Were any new external packages added?

* List external packages added and why they were added

### CODE REVIEW

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Parallel Service Re-architecture
```
Comment (crystalynliu) [inline docs/apidocs/adr/2022-05-12-Redesign.md:1]: move adr folder out of apidocs

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+2. Spike how could we poll events calls based on Jarmo's document, and if these can support current business requirement (general & snapshots)
+  - table by module
+  - table schema
+  - Differences between heiko & para
+3. github actions to deploy and document them
+4. redis cache redis缓存
+5. start polling events/calls one by one (will create ticket at that time)requires a list first
+6. monitoring
+
+
```
Comment (crystalynliu) [inline docs/apidocs/adr/2022-05-12-Redesign.md:102]: remove rest of empty line

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+**### Parallel Service:**
+
+Parallel Service itself has 2 main routers (para/heiko), they both has different versions of endpoints(currently only V1).
+We will add V2 endpoints which will use the new architecture flows to provide data to frontend. During the development of V2, we should make sure V1 can still support frontend stably, so that we can smoothly migrate to V2.
+
+1. Postgres: 2 databases, 1 for para, 1 for heiko (should be done by Ops team). 
+   1. staging: naming para-staging/para-heiko
+   2. prod: naming para/heiko
+2. Redis Design
```
Comment (BruceChar) [inline docs/apidocs/adr/2022-05-12-Redesign.md:35]: Redis also be departed for different network.

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+3. Error handler & Monitoring
+4. Router, Controller, Service
+   1. Router
+   2. Controller: no complex logic
```
Comment (BruceChar) [inline docs/apidocs/adr/2022-05-12-Redesign.md:39]: only handle that things like auth validate and parameters parsing

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+**### Polling Service:**
+1. Postgres: 2 databases, 1 for para, 1 for heiko (should be done by Ops team). 
+   1. staging: naming para-staging/para-heiko
+   2. prod: naming para/heiko
+2. Redis Design
+3. Error handler & Monitoring
+4. Poller, Scanner, Migration
+   1. Poller
+   2. Scanner
+   3. Migration
```
Comment (BruceChar) [inline docs/apidocs/adr/2022-05-12-Redesign.md:51]: we don't need migration any more. Poller is for latest record polling from the subquery and Scanner is scan the record by block height. Maybe we need a Fetcher to fetch the storage by rpc if possible.

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+
+After Polling Service Launched
+1. Set up repo to connect to new PG dbs (heiko & para)
+2. Create V2 endpoints for V1 
+  - create several tickets 
+  - unit test
+  - use the new arch
+3. Test for all V2 endpoints 
+4. Replace V1 to V2 in frontend
```
Comment (BruceChar) [inline docs/apidocs/adr/2022-05-12-Redesign.md:80]: we need to provide the migration doc on api v1 to v2

File: docs/apidocs/adr/2022-05-12-Redesign.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+## Context
+
+As time goes on, we have to support more and more business requirements in parallel service. Plus it depends on lots of SubQuery projects(third party services), which are very unstable. It will unfortunately fail each time SubQuery fails. In order to avoid those scenarios, we decided to migrate current V1 endpoints to V2, and follow new architecture designs to make sure it becomes stable and maintainable
+
+
+## Architecture Designs
+
+**### System Design**
+
+![System Design](./system.png)
```
Comment (BruceChar) [inline docs/apidocs/adr/2022-05-12-Redesign.md:12]: Do we specify the redis for paraller-service and polling-service?

Comment (BruceChar) [reply docs/apidocs/adr/2022-05-12-Redesign.md:12]: The architecture shows that RDS and Redis are isloated, it's untrue, we may add the cache layer for RDS by redis

Comment (BruceChar) [reply docs/apidocs/adr/2022-05-12-Redesign.md:12]: And the direction of arrow in architecture is for data flow or action relation? It's confused that the polling-service have the same flow to RDS, redis and Subquery network.

**Review by crystalynliu — APPROVED** [review]

**Pull request merged**